Prevent pass-through to `get_bloginfo` for `%siteurl%`

jjeaton opened the pull request:
%site_url% is the correct variable that TML is looking for. %siteurl% gets
passed to get_bloginfo and is a deprecated parameter for that function
which throws a notice. This fixes the help text so the correct
variable is recommended.

jfarthing84 commented:
What's with this commit message?

jfarthing84 commented:
Additionally, the new variable should be just `%url%`. It also seems that `%blogname%` should become just `%name%`, even though the former does still work.

jjeaton commented:
@jfarthing84 fixed up the commit message.

> Additionally, the new variable should be just %url%. It also seems that %blogname% should become just %name%, even though the former does still work.

I only updated the help text to reference the default variable here: https://github.com/jfarthing84/theme-my-login/blob/6.4/includes/class-theme-my-login-common.php#L88, (which doesn't map directly to `get_bloginfo`) because the variable recommended in the help text was causing notices.

If you feel like adding new variables, that's fine but `%site_url%` should continue to work for backwards compatibility, otherwise users' emails could break when the plugin is updated.

jfarthing84 commented:
I think I'd rather just add `%siteurl%` to the `$defaults` in `Theme_My_Login_Common::replace_vars`, also leaving the `%site_url%` that is already there.

jjeaton committed:
Prevent pass-through to `get_bloginfo` for `%siteurl%`

The `%siteurl%` email variable was causing a notice as it is a deprecated
parameter for `get_bloginfo`, and it is the variable that was mentioned
in the help text. This adds it as a duplicate of `%site_url%` so that either
will work.

jjeaton commented:
Done. Would love to see these fixes in a new release if possible
